chisel/src/executor.rs

```diff
@@ -0,0 +1,766 @@
+    /// Runs the source with the [ChiselRunner]
+    ///
+    /// ### Returns
+    ///
+    /// Optionally, a tuple containing the [Address] of the deployed REPL contract as well as
+    /// the [ChiselResult].
+    pub async fn execute(&mut self) -> Result<(Address, ChiselResult)> {
+        // Recompile the project and ensure no errors occurred.
+        match self.build() {
+            Ok(compiled) => {
+                if let Some((_, contract)) = compiled
+                    .compiler_output
+                    .contracts_into_iter()
+                    .find(|(name, _)| name.eq(&"REPL"))
+                {
+                    // These *should* never panic after a successful compilation.
+                    let bytecode =
+                        contract.get_bytecode_bytes().expect("No bytecode for contract.");
+                    let deployed_bytecode = contract
+                        .get_deployed_bytecode_bytes()
+                        .expect("No deployed bytecode for contract.");
+
+                    // Fetch the run function's body statement
+                    let run_func_statements = compiled.intermediate.run_func_body()?;
+
+                    // Find the last statement within the "run()" method and get the program
+                    // counter via the source map.
+                    if let Some(final_statement) = run_func_statements.last() {
+                        let final_pc = {
+                            let source_loc = final_statement.loc();
+                            let offset = source_loc.start();
+                            let length = source_loc.end() - source_loc.start();
+                            contract
+                                .get_source_map_deployed()
+                                .unwrap()
+                                .unwrap()
+                                .into_iter()
+                                .zip(InstructionIter::new(&deployed_bytecode))
+                                .filter(|(s, _)| s.offset == offset && s.length == length)
+                                .map(|(_, i)| i.pc)
+                                .max()
+                                .unwrap_or_default()
+                        };
+
+                        // Create a new runner
+                        let mut runner = self.prepare_runner(final_pc).await;
+
+                        // Return [ChiselResult] or bubble up error
+                        match runner.run(bytecode.into_owned()) {
+                            Ok(res) => Ok(res),
+                            Err(e) => Err(e),
+                        }
+                    } else {
+                        // Return a default result if no statements are present.
+                        Ok((Address::zero(), ChiselResult::default()))
+                    }
+                } else {
+                    eyre::bail!("Failed to find REPL contract!")
+                }
+            }
+            Err(e) => Err(e),
```

Comment:
use `?` operator to return the error directly

Reply:
Fixed